include/SampleBuffer.h

```diff
@@ -109,6 +109,10 @@ class LMMS_EXPORT SampleBuffer : public QObject, public sharedObject
 	SampleBuffer( const QString & _audio_file, bool _is_base64_data = false );
 	SampleBuffer( const sampleFrame * _data, const f_cnt_t _frames );
 	explicit SampleBuffer( const f_cnt_t _frames );
+	SampleBuffer( SampleBuffer& orig );
+
+	friend void swap( SampleBuffer& first, SampleBuffer& second ) noexcept;
+	SampleBuffer& operator= ( SampleBuffer& that );
```

Comment:
If you make `m_varLock` mutable, then the copy constructor and assignment operator can take const references again.